rlclientlib/utility/header_authorization.cc

```diff
@@ -0,0 +1,25 @@
+#include "header_authorization.h"
+#include <codecvt>
+
+namespace u = utility;
+
+namespace reinforcement_learning {
+  int header_authorization::init(const utility::configuration& config, api_status* status, i_trace* trace) {
+    const auto* api_key = config.get(name::HTTP_API_KEY, nullptr);
+    if (api_key == nullptr) {
+      RETURN_ERROR(trace, status, http_api_key_not_provided);
+    }
+    _api_key = api_key;
+    #ifdef _WIN32
+    _http_api_header_key_name = u::conversions::utf8_to_utf16(config.get(name::HTTP_API_HEADER_KEY_NAME, value::HTTP_API_DEFAULT_HEADER_KEY_NAME));
+    #else
+    _http_api_header_key_name = config.get(name::HTTP_API_HEADER_KEY_NAME, value::HTTP_API_DEFAULT_HEADER_KEY_NAME);
+    #endif
+    return error_code::success;
+  }
+
+  int header_authorization::insert_authorization_header(http_headers& headers, api_status* status) {
+    headers.add(_http_api_header_key_name, _api_key.c_str());
```

Comment:
although looks like we are trying not to call insert_authorization_header if init wasn't successful, it would be cleaner to return some error code here (http_api_key_not_provided) as well if class was not properly initialized

Reply:
added check